### PR TITLE
Appsec: properly populate event

### PR DESCRIPTION
**Pull request by blotus**
The alert will now contains the individual rules that matched, to allow better analysis.

### CODE REVIEW

**Comment by LaurenceJJones** [comment]
private ip:

```
root@bookworm:/opt/crowdsec# cscli alerts inspect 2 -d

################################################################################################

- ID           : 2
- Date         : 2024-05-14T10:02:38Z
- Machine      : c90cca144e044e6297dded898a0e76da
- Simulation   : false
- Reason       : crowdsecurity/vpatch-env-access
- Events Count : 0
- Scope:Value  : Ip:192.168.121.1
- Country      :
- AS           :
- Begin        : 2024-05-14 10:02:38 +0000 UTC
- End          : 2024-05-14 10:02:38 +0000 UTC
- UUID         : a31a4bd3-233c-42a1-93bb-d14143718a97


- Context  :
╭───────────────┬──────────────────╮
│      Key      │      Value       │
├───────────────┼──────────────────┤
│ matched_zones │ REQUEST_FILENAME │
│ method        │ GET              │
│ target_uri    │ /.env            │
╰───────────────┴──────────────────╯

- Events  :

- Date: 2024-05-14 10:02:38 +0000 UTC
╭───────────────┬─────────────────────────────────╮
│      Key      │              Value              │
├───────────────┼─────────────────────────────────┤
│ id            │ 3256131465                      │
├───────────────┼─────────────────────────────────┤
│ matched_zones │ REQUEST_FILENAME                │
├───────────────┼─────────────────────────────────┤
│ method        │ GET                             │
├───────────────┼─────────────────────────────────┤
│ msg           │ crowdsecurity/vpatch-env-access │
├───────────────┼─────────────────────────────────┤
│ name          │ crowdsecurity/vpatch-env-access │
╰───────────────┴─────────────────────────────────╯
```

fake public test:
```

################################################################################################

- ID           : 3
- Date         : 2024-05-14T10:10:52Z
- Machine      : c90cca144e044e6297dded898a0e76da
- Simulation   : false
- Reason       : crowdsecurity/vpatch-env-access
- Events Count : 0
- Scope:Value  : Ip:1.2.3.4
- Country      : AU
- AS           :
- Begin        : 2024-05-14 10:10:52 +0000 UTC
- End          : 2024-05-14 10:10:52 +0000 UTC
- UUID         : 07a8cd55-b9d3-46f4-a5ad-217ae9ad9d5b


- Context  :
╭───────────────┬──────────────────╮
│      Key      │      Value       │
├───────────────┼──────────────────┤
│ matched_zones │ REQUEST_FILENAME │
│ method        │ GET              │
│ target_uri    │ /.env            │
╰───────────────┴──────────────────╯

- Events  :

- Date: 2024-05-14 10:10:52 +0000 UTC
╭───────────────┬─────────────────────────────────╮
│      Key      │              Value              │
├───────────────┼─────────────────────────────────┤
│ id            │ 3256131465                      │
├───────────────┼─────────────────────────────────┤
│ matched_zones │ REQUEST_FILENAME                │
├───────────────┼─────────────────────────────────┤
│ method        │ GET                             │
├───────────────┼─────────────────────────────────┤
│ msg           │ crowdsecurity/vpatch-env-access │
├───────────────┼─────────────────────────────────┤
│ name          │ crowdsecurity/vpatch-env-access │
╰───────────────┴─────────────────────────────────╯
```

File: pkg/acquisition/modules/appsec/utils.go [hunk 1]
```diff
@@ -64,7 +107,38 @@ func AppsecEventGeneration(inEvt types.Event) (*types.Event, error) {
...
+
+		evtRule.Timestamp = now
+
+		evtRule.Meta = make(models.Meta, 0)
+
+		for _, key := range []string{"id", "name", "method", "uri", "matched_zones", "msg"} {
+
+			switch value := matched_rule[key].(type) {
+			case string:
+				evtRule.Meta = appendMeta(evtRule.Meta, key, value)
```
Comment (buixor) [inline pkg/acquisition/modules/appsec/utils.go:123]: we need to ensure this gets properly stripped if it's too long

**Review by buixor — CHANGES_REQUESTED** [review]
we should ensure that meta are stripped if too long (otherwise they will cause endess retry loop), otherwise lgtm.

**Review by buixor — APPROVED** [review]

**Pull request merged**